### ISSUE TITLE
Improve dense layer

### ISSUE THREAD

**Issue by theolepage** [theolepage/prophecy#27]
- [x] Batch training with mini-batch

---

- [ ] Implement cross entropy cost function
- [ ] Cost/loss average over batch for progress display?

---

- [ ] Regularization

---

- [x] Better initialization of weights and biases
- [ ] Learning rate decay (optimizers, custom function)

---

- [ ] Adam optimizer

**Comment by Cc618** [comment]
### Hi !

For section "Better initialization of weights and biases", can I implement Xavier initialization with an initializer class ?

**Comment by theolepage** [comment]
Hey Celian.
Yes sure, feel free to create a pull request based from `develop` as this feature was planned. 
Please note that the project is being fully refactored at the moment and I am not sure I will have much time to dedicate to the development.
